File: app/src/main/java/com/kgb/k_app/database/ChallengeContract.java
package com.kgb.k_app.database;

import android.provider.BaseColumns;

/**
 * @author Krzysztof Betlej <kgbetlej@gmail.com>.
 * @date 2/21/17
 * @copyright Copyright (c)2017 KGBetlej
 */

public final class ChallengeContract {
    private ChallengeContract() {}

    public static class ChallengeEntry implements BaseColumns {
        public static final String TABLE_NAME = "challenge";
        public static final String COLUMN_NAME = "name";
        public static final String COLUMN_DESCRIPTION = "description";
        public static final String COLUMN_TYPE = "type";
        public static final String COLUMN_RATE = "rate";
    }

    public static class YourChallengeEntry implements BaseColumns {
        public static final String TABLE_NAME = "your_challenge";
        public static final String COLUMN_CHALLENGE_ID = "challenge_id";
        public static final String COLUMN_START_DATE = "start_date";
        public static final String COLUMN_EXPIRE_DATE = "expire_date";
        public static final String COLUMN_PROGRESS = "progress";
    }

    public static class ChallengeTypeEntry implements BaseColumns {
        public static final String TABLE_NAME = "challenge_type";
        public static final String COLUMN_NAME = "name";
    }
}

File: app/src/main/java/com/kgb/kapp/db/ChallengeContract.kt
package com.kgb.kapp.db

import android.provider.BaseColumns

val DATABASE_NAME = "habittrainer.db"
val DATABASE_VERSION = 1

object ChallengeEntry : BaseColumns {
    val TABLE_NAME = "challenge"
    val _ID = "id"

    val TYPE_COL = "challenge_type"
    val STEP_COL = "challenge_step"
    val STEP_PROGRESS_COL = "challenge_step_progress"
}

object ChallengeStepEntry : BaseColumns {
    val _ID = "id"
    val TABLE_NAME = "challenge_name"
    val TYPE_COL = "type"
    val DESCR_COL = "description"
}